src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -311,21 +369,62 @@ public InvocationHandler(Type[] parameterTypes, Action<object[]> handler)
             }
         }
 
-        private struct InvocationRequest
+        private class InvocationRequest : IDisposable
```

Comment:
a) it's going in a dictionary, so it'll be boxed anyway
b) we used to put it in once and take it out once and never look at it in between, now we need to keep pulling it out of the dictionary for each stream item we get (in theory).

This is somewhat premature as it happened in my prototype that did streaming, but I think it's still going to be useful. We can always review when we have perf data.